core/src/ledger/governance/utils.rs

```diff
@@ -161,9 +174,18 @@ impl Display for ProposalResult {
 }
 
 impl ProposalResult {
-    /// Return true if two third of total voting power voted nay
-    pub fn two_third_nay(&self) -> bool {
-        self.total_nay_power >= (self.total_voting_power / 3) * 2
+    /// Return true if at least 1/3 of the total voting power voted and at least
+    /// two third of the non-abstained voting power voted nay
+    pub fn two_thirds_nay_over_two_thirds_total(&self) -> bool {
+        let at_least_two_thirds_voted = self.total_yay_power
+            + self.total_nay_power
+            + self.total_abstain_power
+            >= self.total_voting_power / 3 * 2;
```

Comment:
```suggestion
            >= self.total_voting_power * 2 / 3;
```

in general, we should practice multiply-then-divide

Reply:
I think the operators precedence in Rust already evaluates multiplications before divisions, but I can change it. Do you think at this point it's better to split the fraction among the two sides of the inequality an be `(yay + nay + abstain) * 3 >= total * 2`?

Reply:
how you have it is fine, I just prefer `* 3 / 2`. admittedly this is somewhat of an aesthetic choice